pkg/git/libgit2/managed_test.go

```diff
@@ -442,3 +457,162 @@ func Test_ManagedSSH_HostKeyAlgos(t *testing.T) {
+	g.Expect(err).ToNot(HaveOccurred())
+
+	authOpts := &git.AuthOptions{
+		TransportOptionsURL: getTransportOptionsURL("http"),
+	}
+
+	tmpDir := t.TempDir()
+
+	ctx, cancel := context.WithTimeout(context.TODO(), timeout)
+	defer cancel()
+
+	repoURL := server.HTTPAddress() + "/" + repoPath
+	branch := CheckoutBranch{
+		Branch: git.DefaultBranch,
+		// Set last revision to HEAD commit, to force a no-op clone.
+		LastRevision: fmt.Sprintf("%s/%s", git.DefaultBranch, commit.Id().String()),
+	}
+
+	cc, err := branch.Checkout(ctx, tmpDir, repoURL, authOpts)
+	g.Expect(err).ToNot(HaveOccurred())
+	g.Expect(cc.String()).To(Equal(git.DefaultBranch + "/" + commit.Id().String()))
+	g.Expect(git.IsConcreteCommit(*cc)).To(Equal(false))
+
+	// Set last revision to a fake commit to force a full clone.
+	branch.LastRevision = fmt.Sprintf("%s/non-existent-commit", git.DefaultBranch)
+	cc, err = branch.Checkout(ctx, tmpDir, repoURL, authOpts)
+	g.Expect(err).ToNot(HaveOccurred())
+	g.Expect(cc.String()).To(Equal(git.DefaultBranch + "/" + commit.Id().String()))
+	g.Expect(git.IsConcreteCommit(*cc)).To(Equal(true))
+}
+
+func TestManagedCheckoutTag_Checkout(t *testing.T) {
+	managed.InitManagedTransport(logr.Discard())
+	g := NewWithT(t)
+
+	timeout := 5 * time.Second
+	server, err := gittestserver.NewTempGitServer()
+	g.Expect(err).ToNot(HaveOccurred())
+	defer os.RemoveAll(server.Root())
+
+	err = server.StartHTTP()
+	g.Expect(err).ToNot(HaveOccurred())
+	defer server.StopHTTP()
+
+	repoPath := "test.git"
+	err = server.InitRepo("../testdata/git/repo", git.DefaultBranch, repoPath)
+	g.Expect(err).ToNot(HaveOccurred())
+
+	repo, err := git2go.OpenRepository(filepath.Join(server.Root(), repoPath))
+	g.Expect(err).ToNot(HaveOccurred())
+
+	branchRef, err := repo.References.Lookup(fmt.Sprintf("refs/heads/%s", git.DefaultBranch))
+	g.Expect(err).ToNot(HaveOccurred())
+	defer branchRef.Free()
+
+	commit, err := repo.LookupCommit(branchRef.Target())
+	g.Expect(err).ToNot(HaveOccurred())
+	_, err = tag(repo, commit.Id(), false, "tag-1", time.Now())
+
+	checkoutTag := CheckoutTag{
+		Tag: "tag-1",
+	}
+	authOpts := &git.AuthOptions{
+		TransportOptionsURL: getTransportOptionsURL("http"),
+	}
+	repoURL := server.HTTPAddress() + "/" + repoPath
+	tmpDir := t.TempDir()
+
+	ctx, cancel := context.WithTimeout(context.TODO(), timeout)
+	defer cancel()
+
+	cc, err := checkoutTag.Checkout(ctx, tmpDir, repoURL, authOpts)
+	g.Expect(err).ToNot(HaveOccurred())
+	g.Expect(cc.String()).To(Equal("tag-1" + "/" + commit.Id().String()))
+	g.Expect(git.IsConcreteCommit(*cc)).To(Equal(true))
+
+	checkoutTag.LastRevision = "tag-1" + "/" + commit.Id().String()
+	cc, err = checkoutTag.Checkout(ctx, tmpDir, repoURL, authOpts)
+	g.Expect(err).ToNot(HaveOccurred())
+	g.Expect(cc.String()).To(Equal("tag-1" + "/" + commit.Id().String()))
+	g.Expect(git.IsConcreteCommit(*cc)).To(Equal(false))
+}
+
+func getTransportOptionsURL(transport string) string {
```

Comment:
Wonder whether we should lean on the existing type for transports:
https://github.com/fluxcd/source-controller/blob/main/pkg/git/options.go#L57

```suggestion
func getTransportOptionsURL(transport git.TransportType) string {
```